src/component/mxgraph/shape/IconPainter.ts

```diff
@@ -15,35 +15,23 @@
  */
 
 import { mxgraph } from 'ts-mxgraph';
-import MxScaleFactorCanvas, { MxCanvasUtil } from '../extension/MxScaleFactorCanvas';
+import BpmnCanvas, { MxCanvasUtil } from './render/BpmnCanvas';
 import StyleUtils from '../StyleUtils';
+import { IconStyleConfiguration, Size } from './render/render-types';
 
 export interface PaintParameter {
   c: mxgraph.mxXmlCanvas2D;
   shape: ShapeConfiguration;
-  icon: IconConfiguration;
+  icon: IconStyleConfiguration;
   ratioFromParent?: number;
```

Comment:
To have the same name as **IconConfiguration**:
```suggestion
  ratioFromShape?: number;
```

Reply:
Forget this suggestion, because we need to use the envelope icon for the edge of the message flow; There is no notion of Shape. 
So, maybe, we need to rename the field in **IconConfiguration**.